### ISSUE TITLE
[Due for payment 2025-06-27] [$250] [HybridApp] Improvement to HybridApp TTI / NewDot authentication page

### ISSUE THREAD

**Issue by Julesssss** [Expensify/App#49844]
P/S authored by @staszekscp [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1727439145575909)

### Problem

The TTI on HybridApp is longer than on a standalone NewDot application. In order to determine if a specific user should go to NewDot, HybridApp has to boot OldDot code first. The business logic for OldDot was written in JS in a framework called YAPL, and it runs on JSC.

Potentially also resolves: 
- https://github.com/Expensify/App/issues/56685

### Solution

Let’s replace JSC with Hermes! Hermes is known for its advantage over JSC in initial boot time, and it became a standard in the React Native world. Since YAPL is written in JS, theoretically should be able to do that, and get rid of JSC from the project. As bonus the app size would become smaller, too. :muscle::skin-tone-2: :rocket:

However, I’d like to emphasise that, since YAPL has been written in legacy JavaScript (mostly pre ES2015) there is a chance that the migration may become very time-consuming. Nevertheless I think it is worth to do at least a research, and see how far we can get, because it will speed up the both, OldDot, and HybridApp! 

**Mateusz' updated [Solution](https://expensify.slack.com/archives/C01GTK53T8Q/p1730278677529929)**

Adjust the current authentication flow on HybridApp to replace the OldDot sign-in page with the NewDot version.

1. Currently, we perform authentication on the OldDot side and pass all necessary credentials to enable the New Experience and handle authentication/re-authentication. We could reverse this by using the NewDot sign-in page to generate all required credentials. These credentials could then be used to authenticate on the OldDot side. To trigger authentication on YAPL side, native methods can be used.
2. We also need to decide at the start of sign-in whether the user should be redirected to OldDot or NewDot. The `SignInUser` command should return an additional field with `nvp_tryNewDot`, allowing us to determine based on the value of the `dismissed` key whether to stay on NewDot or transition to OldDot.
3. Another important aspect from the user’s perspective is error handling. OldDot initialisation might fail, so we should be able to display errors on the SignIn page, just as we do when NewDot authentication fails. We can track OldDot initialisation through native events and store the current state in Onyx. Errors returned by OldDot’s API can then be passed via these events and displayed on the NewDot SignIn page

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021933575711850825609</li>
        <li>Upwork Job ID: 1933575711850825609</li>
        <li>Last Price Increase: 2025-06-13</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @</details>

**Comment by staszekscp** [comment]
Hey! With @WoLewicki we'll be working on this issue!

**Comment by Julesssss** [comment]
Work in ongoing.

**Comment by Julesssss** [comment]
In progress

**Comment by Julesssss** [comment]
Hey @staszekscp & @WoLewicki. It sounds like you've been making progress here. 

Let's use this issue to track the plan and the backend changes that you'll need me to implement. I'm currently struggling to find the 3 point plan that was shared recently. I asked [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1730728886409989?thread_ts=1727439145.575909&cid=C01GTK53T8Q) and will update this issue once reviewed.

**Comment by Julesssss** [comment]
I added Mateusz' most recent [Solution](https://expensify.slack.com/archives/C01GTK53T8Q/p1730278677529929) to the issue description. 

In order to return the value of `nvp_tryNewDot` I think I would need to return it from both `SigninUser` and `SigninUserWithLink` (for magic links). I'm going to look into this this afternoon.

**Comment by Julesssss** [comment]
Hey @staszekscp, will it be helpful to start returning`tryNewDot` yet? I have a WIP PR but will hold to check TTI changes are actively being worked on. 

> We also need to decide at the start of sign-in whether the user should be redirected to OldDot or NewDot. The SignInUser command should return an additional field with nvp_tryNewDot, allowing us to determine based on the value of the dismissed key whether to stay on NewDot or transition to OldDot.

**Comment by Julesssss** [comment]
I have started to submit the [WIP PRs ](https://github.com/Expensify/Auth/pull/13016) for the following backend change that will start returning the necessary field:

> We also need to decide at the start of sign-in whether the user should be redirected to OldDot or NewDot. The SignInUser command should return an additional field with nvp_tryNewDot, allowing us to determine based on the value of the dismissed key whether to stay on NewDot or transition to OldDot.

**Comment by war-in** [comment]
Hi :wave:

While working on the new sign-in page for HybridApp, I encountered a situation I'd like your input on.

When the nvp is set to open ND, we sign the user into the new app and simultaneously attempt an OD sign-in in the background to expedite the process. However, the OD sign-in can occasionally fail, leaving us in a state where the ND app is active, but OD is not signed in.

If the user decides to switch back to the classic experience, we must reattempt the OD sign-in, which can take a few seconds.

Given this scenario, what would be the best user experience? Should we display a new page notifying them of the ongoing switch, or (my personal preference) show a loading indicator next to the button (as shown in the screen below)?

Any input is highly appreciated :pray:

![image](https://github.com/user-attachments/assets/819269df-ca76-421b-90de-ebdf0eec1801)
cc @Expensify/design

**Comment by dannymcclain** [comment]
>show a loading indicator next to the button (as shown in the screen below)

@war-in are you talking about something like this screenshot? (Grabbed it from the related Slack thread)

![image](https://github.com/user-attachments/assets/4594ff49-e9c2-4f9c-b4f5-88fb342a96c4)

I think something like that is good—and I think we agree it'd be a lot nicer to just add a little loading indicator than take the user to a new page. Let's just make sure we're using our standard spinner/loader component for that little guy.

**Comment by Julesssss** [comment]
Hey @dannymcclain, I'm struggling to locate an example of a loading spinner in Figma or within the codebase, would you mind sharing an example so we can find the existing component? Thanks

**Comment by dannymcclain** [comment]
I'm honestly not sure that we have super good examples in Figma. And I know it's appearance changes a little bit between web and native apps... 

[This](https://www.figma.com/design/qQIIaxVUdtggrh8eg8bI5B/Brand-Guidelines-%26-Product-UI?node-id=4502-9332&node-type=frame&t=faQlLcJ1jKbguhQU-11) is what we have in Figma currently (look for `Loader / Spinner`):

![CleanShot 2024-11-12 at 08 23 08@2x](https://github.com/user-attachments/assets/448692e3-cdd2-4abe-b4fa-a1dc55e9f9f4)


@Expensify/design can either of you think of any recent examples of us implementing the spinner that Jules could check out? I'm coming up blank!

**Comment by Julesssss** [comment]
Ah yes I immediately remember that now I see it 😄

I think [FullScreenLoadingIndicator](https://github.com/Expensify/App/blob/main/src/components/FullscreenLoadingIndicator.tsx#L17)  is where we would start.

**Comment by war-in** [comment]
> It might even be the default React native ActivityIndicator component

Actually, I used `ActivityIndicator` in my example but I'll double check before creating a PR

**Comment by Julesssss** [comment]
Oh, cool, maybe it's something to do with `theme.spinner`. Anyway, let us know if you need any more help

**Comment by war-in** [comment]
And another question on the `Switch to Expensify Classic` button.

What should we do when the API call fails again after clicking the button? Should we show some error message under the button? 

<img width="372" alt="image" src="https://github.com/user-attachments/assets/ded361b8-2221-444d-9ab5-f47884e8ec66">

Or maybe an error icon?

<img width="371" alt="image" src="https://github.com/user-attachments/assets/9f174def-d586-43d3-994a-3b250770d574">

However, both have disadvantages:
- error message:
  - The `MenuItem` component is not prepared to handle the error messages correctly (at least in this place) which results in strange behaviour ⬇️, so some additional work is required to show it properly

https://github.com/user-attachments/assets/0a42f04e-f1c7-44fc-961c-7c860d4c2359

- error icon
  - it shows that something happened but doesn't provide information about possible solutions, so we would have to think of some other way (modal?, tooltip?) of showing the error
  
Please share your thoughts 🙏 

cc @Julesssss @dannymcclain

**Comment by dannymcclain** [comment]
Going to cc in the @Expensify/design team for thoughts on this. I wonder if it would be better to just pop a little modal if there's an error letting the user know what happened.

**Comment by shawnborton** [comment]
Either that, or some kind of dismissible RBR error message right under the option row could work too? That would be pretty consistent with the rest of NewDot.

**Comment by dannymcclain** [comment]
Yeah I could see that too. 🤔

Here's a dismissible error in the LHN and a modal. How are we feeling @Expensify/design?

![image](https://github.com/user-attachments/assets/400354de-0c51-4c4d-9132-43bb2ed09c75)

**Obviously** we would get the proper copy for either approach. 😅

**Comment by shawnborton** [comment]
Thanks for sharing these. After seeing both, it feels like the centered dialog feels more appropriate here? What do you think?

**Comment by dubielzyk-expensify** [comment]
+1 to dialog

**Comment by Julesssss** [comment]
Not overdue, backend PRs are awaiting deployment

**Comment by Julesssss** [comment]
Initial backend PRs are deployed and I have been working with @war-in to help move the auth refactor forward. We noticed a couple of issues which will need to be resolved:

- The App client doesn't force thee correct tryNewDot dismissed type, which is causing bugs on the backend (Force correct type for TrynewDot NVP classicRedirect.dismissed)
- A separate issue is setting the type incorrectly on the backend, which needs to be resolved ([issue](https://github.com/Expensify/Expensify/issues/447548))

**Comment by Julesssss** [comment]
I tested early local builds today:
- `cd Mobile-Expensify`
- `git fetch && gh pr checkout 13284`
- `cd react-native && gh pr checkout 51257`

It's working nicely! (Android prod build and iOS dev). I haven't tested a prod iOS build yet. The only issue I noticed was with the slight flickering on the Ilaunch Activity/Splash screen transition: 

https://github.com/user-attachments/assets/0c42732c-2d46-49f3-b49d-db20d8d99554

**Comment by Julesssss** [comment]
I spoke with @war-in today about necessary backend changes. I have detailed them and will working on this task [here](https://github.com/Expensify/Expensify/issues/450290).

**Comment by Julesssss** [comment]
Changes are in progress. Focus is temporarily on the big [HybridApp repo changes](https://github.com/Expensify/App/issues/49845).

**Comment by Julesssss** [comment]
Focus remains on these HybridApp dev environment [issues](https://github.com/Expensify/App/issues/49845#issuecomment-2553411188) for now

**Comment by Julesssss** [comment]
Just sharing an updated overview now that I am free to work on the backend changes: 

- This is the main tracking issue for NewDot startup
- Initial changes allowing data to be sent from are in review here ([OD](https://github.com/Expensify/Mobile-Expensify/pull/13416) & [ND](https://github.com/Expensify/App/pull/56698)) - [issue](https://github.com/Expensify/App/issues/56836)
- I am now able to start working on [these](https://github.com/Expensify/App/issues/53556) backend changes to unblock further changes
- Further progress on the ND authentication page to be made

**Comment by Julesssss** [comment]
Backend is now [providing](https://github.com/Expensify/App/issues/53556#issuecomment-2751779970) tryNewDot onyxData and in the response data in some cases which should unblock work here.

**Comment by Julesssss** [comment]
I'm fixing the incorrect NVP key [here](https://github.com/Expensify/Web-Expensify/pull/46125) -- I'll share here when it hits staging/prod.

**Comment by Julesssss** [comment]
Hey @war-in, just checking in to see if this is still in progress?

**Comment by war-in** [comment]
Hey, yes! We're actively hunting bugs
@289Adam289 could you elaborate a bit? 🙏

**Comment by 289Adam289** [comment]
Hi, the main sign-in flow works pretty good I am currently working on Onboarding flow for new accounts. For some new accounts the onboarding flow doesn't start. Unfortunately, this behavior seems quite random. 

Once we fix that I believe we can move on to other sign-in flows: SAML, Apple and Google.

**Comment by 289Adam289** [comment]
Hi,
@Julesssss  I've managed to solve most of the main flow problems, but still there are some bugs and questions that would require your attention.
1. I believe that backend returns wrong `nvp_tryNewDot` value for new accounts both created on ND and classic expensify:
When creating new account `BeginSignIn` request returns:
```
{
        "onyxMethod": "merge",
        "key": "nvp_tryNewDot",
        "value": []
}
```
Same thing happens when trying to log in with account created on expansify.com. Because it's an empty array following (correct) onyx merge updates do not overwrite it.

2. In scenario:
 - create account on classic web Expensify
- open hybrid app
- log in to the account
- We are immediately moved to OD

Should the `Try New Expensify` button be visible? I assume that it should because we want users to try the new app.

3.  I am moving my focus to rest of the sign in flows. Is there any way to test Google and Apple flows without adhoc builds? Or do we have to relay on them.

cc @mateuuszzzzz @war-in

**Comment by Julesssss** [comment]
> I believe that backend returns wrong nvp_tryNewDot value for new accounts both created on ND and classic expensify:

Ah so that is the correct value of the NVP, it is empty during `beginSignIn` and set when account is opened depending on the email domain and the site you are using to authenticate. It might remain empty or be set during `signUpUser`. This would be problematic to change.

> 2. We are immediately moved to OD

You shouldn't be redirected in this case, you will only be redirected to New Expensify from OldExpensify. However, if you manually navigated to OldDot the `Try New Expensify` button will only be displayed if the user has previously be set to the new experience. So if classicRedirect.dismissed is either true or false they will see the button. If the NVP is an empty array (as you saw above), then the user won't see the button (we only want some users to try new dot currently).

3. I think AdHoc build is the main way to test. Google SSO _should_ work on AdHoc builds, but I'm not sure that Apple SSO is testable. I can look into this though. Presumably there is a method.

**Comment by 289Adam289** [comment]
> Ah so that is the correct value of the NVP, it is empty during beginSignIn and set when account is ...

Oh ok, I understand, but would it be possible to an change empty array to an empty object. I don't think it is possible to overwrite an empty array with an object in Onyx using merge. This causes the value of `nvp_tryNewDot` locally to be always equal to `[]`, or at least until Onyx is cleared.

> You shouldn't be redirected in this case, you will only be redirected to New Expensify from OldExpensify.

Do  I understand correctly that new accounts (even created on expensify.com) should always go to ND?

**Comment by Julesssss** [comment]
> Oh ok, I understand, but would it be possible to an change empty array to an empty object. I don't think it is possible to overwrite an empty array with an object in Onyx using merge.

Ah gotcha, yeah that should be fine. I'll change that soon.

> Do I understand correctly that new accounts (even created on expensify.com) should always go to ND?

So this is changing frequently as we push more and more users to NewDot. Here is the most up to date logic: 

> All users signing up via NewDot will have the redirect NVP applied
> Users signing up via OldDot on public domains will have the redirect NVP applied

**Comment by trjExpensify** [comment]
>Users signing up via OldDot on public domains will have the redirect NVP applied

Mhm, is that accurate? 

Anyone with any email that signs up on expensify.com and chooses individual or 1-10 is redirected into NewDot. Very soon though, that will be anyone who chooses 10+ as well I believe (CC: @chiragsalian). So all new sign-ups will go to NewDot, and then anyone identifying as 50+ in the onboarding modal will be sent back to Classic. I think this is the App issue to get it over the line: https://github.com/Expensify/App/issues/60161

On HybridApp right now, all sign-ups go to the NewDot experience. Those that choose a 11+ option are being redirected back to Classic, but I think this will become 50+ as well soon per the above.

**Comment by chiragsalian** [comment]
> I think this is the App issue to get it over the line: https://github.com/Expensify/App/issues/60161

The issue to get it over the line is https://github.com/Expensify/Expensify/issues/490399. I'm just waiting for some other parts to get live before moving https://github.com/Expensify/Expensify/issues/490399 to live.

**Comment by trjExpensify** [comment]
Yeah cool, I'm not sure if this issue is talking about the mobile app only, and specifically. In which case, all mobile sign-ups are  going into NewDot already -- we're just sending anyone that chooses one of the 10+ options in the companyStep back to Classic -- which I believe we'll stop doing when the accountingStep upgrade comes in?

**Comment by Julesssss** [comment]
This issue is regarding the HybridApp specific authentication page refactor. But good point Tom yeah, for mobile you are taken to NewDot for new accounts, BUT if you self select as a team with more than 10 users you'll be switched back to OldDot. I'll update [this](https://stackoverflowteams.com/c/expensify/questions/19269).

**Comment by Julesssss** [comment]
Progress ongoing.

**Comment by 289Adam289** [comment]
Hi,
On more update on our side.

I've worked on Google and Apple flow configuration with @mateuuszzzzz. We have reused OD configuration which should  work properly. At least on release builds. Unfortunately, not every flow works with AdHoc builds even on main, which makes it very hard to verify that the configuration provided by us is correct. In current state there are no errors returned by Google or  Apple flows. The only error we get is I suspect returned from BE. The message is as follows:

```
Cannot get account details, please try again or contact concierge@expensify.com
```
It's problematic to tell whether it's caused by wrong token format or some BE issue.

Is it possible for internal engineers to maybe test the changes on local signed release build and verify the configuration is correct? A lot of the changes in this pr are in native code that runs before we even get information from API. That makes it impossible to hide them behind the beta.

Testing both flows and BE issue with `[]` are the main blokers right now. Apart from that PR is really close to be ready for review.

**Comment by Julesssss** [comment]
Hey, I just got back from holiday. Nice progress 🎉 

> It's problematic to tell whether it's caused by wrong token format or some BE issue.

I'm certain that Google SSO used to work on **Android** AdHoc builds and I can see they do not with the current AdHoc build on your PR, so hopefully it is a new BE issue I can help with. Apple will be trickier, I'm not sure we ever had a real method of testing this.

> Backend changes (changing [] to {} when nvp_tryNewDot is empty)

I'll work on this next. I just need to catch up with some things first.

**Comment by 289Adam289** [comment]
@Julesssss I've installed Android AdHoc build from a different PR and I wasn't able to login using Google flow.

Let me know if theres anything more I can do to move this pr forward.

**Comment by Julesssss** [comment]
> I've installed Android AdHoc build from a different PR and I wasn't able to login using Google flow.

Alright thanks, I think it broke recently so I'll take a look at this next. Today I submitted the PR which make the NVP default to an empty object. It should be live in a few days.

**Comment by Julesssss** [comment]
Thanks @289Adam289. 

> There are still some flows that need to be tested once the BE changes are ready for example Onboarding flow.

The backend changes are now live on staging and should hit prod by tomorrow. 

In the meantime I'm looking for methods of testing SSO on AdHoc builds.

**Comment by Julesssss** [comment]
Functionality is working well on the [latest](https://github.com/Expensify/App/pull/51257#issuecomment-2916873874) AdHoc build! I noticed a couple of minor issues when testing:
-[Android] Notification permission prompt occurred when opening the app, not after authentication

[Android | iOS] I briefly saw the Auth page input field when signing out, it then switched back to the email field - not a blocker though
I can't share the full details here, but I'm looking into AdHoc SSO in the meantime.

**Comment by jponikarchuk** [comment]
# Android - App returns to Join page briefly after signing out

If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** v9.1.53-5 PR:51257
**Reproducible in staging?:** Yes
**Reproducible in production?:** No
**If this was caught during regression testing, add the test name, ID and link from TestRail:** Exp
**Email or phone of affected tester (no customers):** ddksskskls@gmail.com
**Issue reported by:** Applause Internal Team
**Device used:** Samsung Galaxy Z Fold 4 / Android 14
**App Component:** undefined

## Action Performed:
1. Launch adhoc app.
2. Enter email > Continue.
3. Tap Join.
4. Complete onboarding.
5. Go to Account.
6. Tap Sign out.

## Expected Result:
App will return to login page.

## Actual Result:
App returns to Join page in Step 3 briefly after signing out.

The same issue happens when logging in with existing account and then sign out. App will return to magic code page briefly.

## Workaround:
Unknown

## Platforms:
- [x] Android: App
- [ ] Android: mWeb Chrome
- [ ] iOS: App
- [ ] iOS: mWeb Safari
- [ ] iOS: mWeb Chrome
- [ ] Windows: Chrome
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/d80f3c14-edd3-4828-bd50-e9bbd9900123

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by 289Adam289** [comment]
> App shows banner on the top after cancelling
This is on me. I've added growls to help with debugging on adHoc builds (logs are not visible). I will remove them once the flows are confirmed to be working correctly

**Comment by mateuuszzzzz** [comment]
> [Android] Notification permission prompt occurred when opening the app, not after authentication

@Julesssss could you provide video showing this issue? I am not able to reproduce it.

**Comment by mateuuszzzzz** [comment]
@Julesssss @trjExpensify  I investigated [issue](https://github.com/Expensify/App/issues/49844#issuecomment-2917612772) with notification permission prompt on Android.

We enable permissions twice in [CustomAirshipExtender.java](https://github.com/Expensify/Mobile-Expensify/blob/24d8ef3aacb3a39107f6949c222bd67c6a48e60f/Android/src/org/me/mobiexpensifyg/CustomAirshipExtender.java#L23-L30).

I’m wondering if enabling push notifications in HybridApp is intentional, or just a leftover from the pure OldDot app. 🤔

I tried removing the two lines that enable push notifications, and it worked. The prompt now only shows after sign-in. However, this also prevents users on OldDot from enabling notifications within the app, since on Android the OldDot app doesn’t have the 'Receive real-time alerts' section in settings. 

What are the next steps? Should we include any form of this fix in the SignInPage PR?

**Comment by trjExpensify** [comment]
>since on Android the OldDot app doesn’t have the 'Receive real-time alerts' section in settings.

Interesting. So you're saying the"Receive real-time alerts" in-app OldApp setting exists on iOS, but not Android? Kinda' weird, I wonder why? (CC: @AndrewGable as he might have some historical knowledge). 

I do think it's a bit of a pointless in-app setting really. If you want to manage "all on or off" push notifications, you can do so at the OS level. It's not like we have some additional fine-grained controls like "only notify me about `@mentions` and DMs" etc to tailor user preference. To that end, I'd actually be comfortable with removing "receive real-time alerts" from the OldApp settings menu - especially if that fixes the bug of prompting to enable on first sign-in consistently across all platforms. 

CC: @JmillsExpensify

**Comment by JmillsExpensify** [comment]
> To that end, I'd actually be comfortable with removing "receive real-time alerts" from the OldApp

Agreed

**Comment by Julesssss** [comment]
> I’m wondering if enabling push notifications in HybridApp is intentional, or just a leftover from the pure OldDot app. 🤔

Likely the latter. Also looking at that code it also seems to be an oversight, given there is no accompanying comment.

> ...However, this also prevents users on OldDot from enabling notifications within the app, since on Android the OldDot app doesn’t have the 'Receive real-time alerts' section in settings.

@mateuuszzzzz correct me if I'm wrong, but I understood this to mean that we would no longer prompt OldDot users to enable all notifications? So New Android users on OldDot (or existing users changing devices) won't get notifications until they are migrated to NewDot or manually navigate to settings. It's not terrible, but we should attempt an OldDot fix regardless I think.

**Comment by Julesssss** [comment]
Separately I have been making progress with the broken SSO triage. Details shared internally [here](https://github.com/Expensify/Expensify/issues/505917#issuecomment-2923030085).

**Comment by 289Adam289** [comment]
#### Update:

All the bugs found by QA have been fixed (@mateuuszzzzz will add changes fixing notification today). 
So the only problems left are Google and Apple flows. I believe @Julesssss is working on it. Let me, know if there is anything we can do to help with the remaining flows.

I will remove the growls once the flows are ready.

**Comment by Julesssss** [comment]
> Let me, know if there is anything we can do to help with the remaining flows.

The only thing I can think of is to resolve the lint issues. 

Yeah I'm making some backend changes which should get us closer. I'm waiting on deployments and will share updates asap.

**Comment by mateuuszzzzz** [comment]
Hi! As @289Adam289 mentioned, I’ve made the adjustments to the notifications. I think we can go ahead and start another round of QA on this PR to check if the previously mentioned issues have been resolved.

I also removed real-time updates toggle from settings on OldDot.

**Comment by Julesssss** [comment]
Updates have been shared daily internally [here](https://github.com/Expensify/Expensify/issues/505917). I shared a public update [here](https://expensify.slack.com/archives/C07HPDRELLD/p1748968662887079?thread_ts=1748915035.997109&cid=C07HPDRELLD) today. More Auth changes need to be deployed, then the first fix can be tested to give us a more information.

**Comment by Julesssss** [comment]
Hey @289Adam289 @war-in @mateuuszzzzz. Google SSO should now work for you on iOS/Android AdHoc builds. I can also sign in with Apple SSO on **Android**, but **not iOS**. I can look further into this when I'm back on Thursday. 

One issue I noticed is that following SSO I'm always routed to OldDot, even if my NVP is set to `classicRedirect: dismissed: false`. I haven't had a chance to look yet, but let me know if the server isn't returning this correctly. 

One other note. It seems to me that the token we pass from mobile doesn't include  `partnerUserID`, like web does. I'm not sure this is a real problem yet, but if you had a moment to verify this from the front end side that would also be helpful.

**Comment by mateuuszzzzz** [comment]
> One other note. It seems to me that the token we pass from mobile doesn't include partnerUserID, like web does. I'm not sure this is a real problem yet, but if you had a moment to verify this from the front end side that would also be helpful.

Hmmmm can `partnerUserID` be associated with `autoGeneratedLogin`? We handle authentication on NewDot side now so once we complete sign-in we obtain `autoGeneretad{Login,Password}`. OldDot uses those values as well so we don't need to pass this value from OldDot to NewDot. 

EDIT: We have this line on NewDot in `Authentication.ts`: `partnerUserID: credentials?.autoGeneratedLogin`.  So I assume this is just a different name for the same thing.

**Comment by Julesssss** [comment]
I generated builds today [here](https://github.com/Expensify/App/pull/51257#issuecomment-2945876971), but noticed a new issue on both AdHoc and local builds. I shared the screen captures privately [here](https://github.com/Expensify/Mobile-Expensify/pull/13284#issuecomment-2946040328).

- Sign in with Google/Apple SSO
- Complete the flow, you land back at the authentication page
- Enter email in manual field
- Hit continue
- Signed into app without entering password

Also, Apple Sign for iOS devices should be fixed with the next Auth deployment (Monday latest).

**Comment by 289Adam289** [comment]
#### Update

I've also noticed bug mentioned by @Julesssss and I am currently working on it.

All the flows are handled in pretty much the same way. In every flow we wait for BE `onyxData` credentials and `tryNewDot`. Unfortunately even though BE returns correct information in Both Apple and Google flows `nvp_tryNewDot` key is not updated in Onyx (`useOnyx` returns undefined and `Onyx.connect` is not triggered). This leaves app in weird state: user is logged in but app is unable to decide whether go to OD or ND. 
For now I see two possible root causes of this behavior:
- The value is overwritten immediately after api request
- The `nvp_tryNewDot` value is lost somewhere between correct api response and actual `Onyx.merge`.

**Comment by 289Adam289** [comment]
#### Update 

We've managed to fix all issues regarding Apple and Google flows. I've tested it on dev builds (Apple flows works well on android) an everything seems to be fine, but it requires confirmation. Once the AdHoc builds are ready I will thoroughly tests both flows.  The root cause of reported issue was logic in file `QueuedOnyxUpdates.ts` that removes all onyx data (except preservedKeys array) before `accountID` is available.

One more thing I've noticed is that on dev and adhoc builds reopening closed accounts moves user straight to OD. `Try New Dot` button is not visible which  suggests that `nvp_tryNewDot` is empty. Is this expected behavior? The reproduction steps:
1. create account on ND sign-in page
2. Close the account
3. User is moved to Sign in page
4. Login to the closed account again
5. Insert magic code
6. User ends up in OD and TryNewDot button is not visible.

@Julesssss for thoughts

Edit:
iOS:
Google SSO - works well 💯 
Apple - Does not work. Api error `Couldn't get account details ...`


Android: 
I wasn't able to test because of failing builds, but Apple Sign in works well on dev builds

**Comment by Julesssss** [comment]
Thanks Adam. Today the iOS:Apple SSO fix will be deployed (**DONE**) and once Jitpack is up we should be able to verify all 4 variants 🎉 

> One more thing I've noticed is that on dev and adhoc builds reopening closed accounts moves user straight to OD. Try New Dot button is not visible which suggests that nvp_tryNewDot is empty. Is this expected behavior

No, we'll need to resolve that. Though I don't think it should block the refactor as it's [pre-existing.](https://github.com/Expensify/App/issues/63506)

**Comment by Julesssss** [comment]
We're looking good. Lets get the PRs ready for final review 🤞 

**Plan for tomorrow:** 
- @289Adam289 and team to confirm Apple SSO on iOS is now working
- @289Adam289 and team to resolve final ESLint issue with App[ PR](https://github.com/Expensify/App/pull/51257)
- @289Adam289 and team to add multiple test cases to the App PR. Then we can link to them from Mobile-Expensify PR
- @Julesssss to pass AdHoc build to QA for second round of testing
- @Julesssss to run additional testing, and to collect additional test cases for QA

**Comment by 289Adam289** [comment]
I've verified with @mateuuszzzzz That all SSO flows work correctly on AdHoc builds🥳 
I've commented on the EsLint issues and removed growls.
~The test cases will be added later today.~ Test cases are on ND PR.

Regarding the QA I think because of how big the PR is it would be nice to test all the flows and run full QA on this pr. There are many edge cases and we might have missed some.

**Comment by Julesssss** [comment]
Seeking QA testing [here](https://expensify.slack.com/archives/C9YU7BX5M/p1749595886626859). We'll cover as much as possible, but we'll perform the full test suite once it hits staging due to QA time that would take in isolation.

**Comment by Julesssss** [comment]
Discussing results from testing [here](https://github.com/Expensify/App/pull/51257#issuecomment-2968310586) in the first PR. We'll need to resolve some, but others will be created as follow up improvements.

**Comment by 289Adam289** [comment]
The urgent issues are resolved. Only follow ups left.

**Comment by Julesssss** [comment]
@DylanDylann I'm happy to keep you assigned as reviewer, but please review as a higher priority than your other issues on Monday, thanks.

**Comment by Julesssss** [comment]
On track for this week:
- PRs are halfway through final review
- QA has verified that issues raised in second round of testing are fixed
- C+ review in progress

**Comment by Julesssss** [comment]
Ready to merge after a final conflict resolution

**Comment by Julesssss** [comment]
A new issue was introduced with an unrelated change. We have a fix and will resolve this in a solo PR, here's the [plan](https://expensify.slack.com/archives/C07HPDRELLD/p1750183138926489?thread_ts=1750121985.641769&cid=C07HPDRELLD).

The PRs have been merged 🎉

**Comment by Julesssss** [comment]
The feature and language string fix are still awaiting deployment due to a separate issue on staging. Then QA will run the full test suite so we'll keep an eye out for bugs.

**Comment by Julesssss** [comment]
For anyone noticing a regression related to these changes, we need to avoid a revert due to the scale of the huge https://github.com/Expensify/Mobile-Expensify/pull/13284 and https://github.com/Expensify/App/pull/51257 PRs. 

@289Adam289, @war-in @mateuuszzzzz and myself (back Friday) will be able to help you resolve these issues. For any non-blockers, please add them https://github.com/Expensify/App/issues/64348.

**Comment by lakchote** [comment]
Possibly found a regression coming from [this](https://expensify.slack.com/archives/C01GTK53T8Q/p1750349728810619?thread_ts=1750345485.362629&cid=C01GTK53T8Q)

**Comment by lakchote** [comment]
^ The one above got solved.

But we need to fix this one: https://github.com/Expensify/App/issues/64622

cc @289Adam289 @war-in @mateuuszzzzz

**Comment by Julesssss** [comment]
Thanks @lakchote. Both are resolved.

**Comment by Julesssss** [comment]
I'm sure SSO was working this morning, but I'm now unable to use Google SSO on Android 😕 

Raised a blocker [here](https://github.com/Expensify/App/issues/64690) to investigate, and I'll make sure we don't deploy before this is resolved.

**Comment by Julesssss** [comment]
After deploying to prod we noticed [this issue](https://github.com/Expensify/App/issues/64690) for Google SSO on Android. I didn't have much luck debugging it on Friday so to avoid delaying deploys the feature has been reverted. We'll resubmit the changes soon. Minor issues were listed [here](https://github.com/Expensify/App/issues/64348), though these can all be follow up issues.

**Comment by Julesssss** [comment]
We've continued testing and triage [here](https://github.com/Expensify/App/issues/64690). 

[PRs](https://github.com/Expensify/App/pull/64786) are prepped for another release attempt, and we have verified the feature is working in release environment.

**Comment by Julesssss** [comment]
PRs prepped for second attempt. I have asked for pre-merge testing from QA [here](https://expensify.slack.com/archives/C9YU7BX5M/p1750874643612549).

**Comment by sonialiap** [comment]
@Julesssss was the first attempt a regression?

**Comment by Julesssss** [comment]
> [@Julesssss](https://github.com/Julesssss) was the first attempt a regression?

Hey, no. Lets do full payment.

**Comment by Julesssss** [comment]
Today I got the blocking PR merged, then tested and merged the double Signin page PRs. Things looked good on a local release build and we'll keep an eye out for any new issues raised on the checklist.

**Comment by Julesssss** [comment]
@neil-marcellini ah that's unfortunate. We just remerged the HybridApp signin page to main yesterday, though personally I would hold another week to ensure further changes and fixes on our side are applied correctly. The solution will remain very similar though. We mostly reverted the revert commit and made some improvements.

**Comment by Julesssss** [comment]
We're back to prepping for merge after this was reverted. The SWM team are working on [known issues](https://github.com/Expensify/App/issues/64348) and I'm working on [this](https://github.com/Expensify/App/issues/65365) issue with Google SSO that only occurs on Google-signed builds. [Here](https://github.com/Expensify/App/pull/65178) is the PR for attempt 3.

**Comment by Julesssss** [comment]
This is no longer on hold for the Google SSO [issue](https://github.com/Expensify/App/issues/56685), as we have the solution. 

So now we just need to work through the in-progress PRs we opened to split up these changes to be more managable. These issues are: 

**[SignIn Page refactor - unified HybridApp deep-linking](https://github.com/Expensify/App/issues/65679)**
- https://github.com/Expensify/App/pull/65506
- https://github.com/Expensify/Mobile-Expensify/pull/13616

[HybridApp SignIn page - Send hybridAppSettings via native method](https://github.com/Expensify/App/issues/65684)
- https://github.com/Expensify/App/pull/65442
- https://github.com/Expensify/Mobile-Expensify/pull/13614

Then we'll get the main changes merged again: 

**Bring back signin page again**
- https://github.com/Expensify/App/pull/65178
- https://github.com/Expensify/Mobile-Expensify/pull/13607

**Comment by Julesssss** [comment]
- [HybridApp SignIn page - Send hybridAppSettings via native method](https://github.com/Expensify/App/issues/65684)
    - ✅ Completed, both PRs merged
- [HybridApp SignIn page - Add HYBRID_APP onyx key](https://github.com/Expensify/App/issues/65786)
    - ✅ Completed, both PRs merged
- [SignIn Page refactor - unified HybridApp deep-linking](https://github.com/Expensify/App/issues/65679)
    - In testing

**Comment by Julesssss** [comment]
@war-in raised a point about our initial onyxData response not working as expected. I raised a [PR](https://github.com/Expensify/Web-Expensify/pull/47811) to remove `tryNewDot` from the `BeginSignIn` command in as this will take a few days to hit prod. In the meantime will look into solutions where existing user data can be returned correctly for saml and SSO and if this is possible we can adjust our HybridApp plan.

**Comment by Julesssss** [comment]
Hey @war-in, `BeginSignIn` no longer returns the tryNewDot NVP on main. Please let me know if that unblocks you, it'll hit prod in about 20 hours.

**Comment by Julesssss** [comment]
@war-in so during todays testing I verified that Google signed builds require the clientID below. I think we can drop the conditional clientIDs (`CONFIG.IS_HYBRID_APP ?`) and stick with the following. Could you please drop that condition, resolve any lint issues unrelated to the double-PR, and then we just need to verify iOS works with that web clientID. 

Web clientID: `240677659774-86pov3adub93cv4b8uj13g7varolmk2l.apps.googleusercontent.com`

**Comment by Julesssss** [comment]
Hey @war-in, yeah that's correct. I'll test this out today 👍

**Comment by Julesssss** [comment]
The reduced remaining changes have passed early testing, have been merged, and will soon be deployed to staging for further testing.

**Comment by Julesssss** [comment]
Okay so even with out pre-merge testing more blockers were found on staging. We resolved a few, but enough remain that we'll need to revert again 😢 

**Blockers**
- https://github.com/Expensify/App/issues/67949 - the worst, but also not reproducible 😕 
- ~~https://github.com/Expensify/App/issues/67812~~ - fixed ✅  but a non-blocking issue remains:
    - https://github.com/Expensify/App/issues/67953
- https://github.com/Expensify/App/issues/67939
- ~~https://github.com/Expensify/App/issues/67814~~ - fixed ✅ 
- https://github.com/Expensify/App/issues/67848
https://github.com/Expensify/App/issues/67920

**Comment by Julesssss** [comment]
@war-in as we discussed earlier today we'll need to revert unfortunately as a few more issues were discovered. Could you please prepare revert PRs? I'm hoping that @mountiny will kindly help with a test build and the CP, but else I'll handle them first thing PT time. 

As for the next steps, I'm most worried about https://github.com/Expensify/App/issues/67949, which I can't reproduce -- but may occur when a token is expired when the user gets the update. I recall we had a similar issue early in HybridApp development.

**Comment by Julesssss** [comment]
**Update**

Main issues blocking re-release: 
- [User signed out when updating to new auth ](https://github.com/Expensify/App/issues/67949#issuecomment-3178457864)
    - @mateuuszzzzz making great progress here, the client migration plan is in development
- [Sign in page text not displayed correctly after switching experiences](https://github.com/Expensify/App/issues/67953)
    - @war-in fixed and it's awaiting deployment
- https://github.com/Expensify/App/issues/67939
    - To be verified when main PRs are ready for full QA testing
- https://github.com/Expensify/App/issues/67848
    - To be verified when main PRs are ready for full QA testing

**Comment by Julesssss** [comment]
Testing went well! There's one issue to resolve then I think we're ready for QA to test again. Full results and screen captures [here](https://github.com/Expensify/App/pull/67996#issuecomment-3185814318).

**Comment by sonialiap** [comment]
Since this is still in progress, i've removed the `awaiting payment` label

**Comment by Julesssss** [comment]
The feature PRs are testing well and I have asked QA to run through our test cases once before we merge the feature.

**Comment by Julesssss** [comment]
We're just about to merge the feature branches. I'm just waiting for the current checklist to ship to give us a bit longer. We're also [blocked](https://expensify.slack.com/archives/C03TQ48KC/p1755722907801739) by our 1pass rate limit... I should be merging tomorrow.

**Comment by Julesssss** [comment]
The Sign-in feature [tested great](https://github.com/Expensify/App/pull/67996#issuecomment-3221814307) for me today with the new fixes 🎉 

I have merged the feature and it'll hit staging with the next App release (hopefully today/tomorrow). We'll keep on top of any new issues that are noticed with the full regression test suite.

**Comment by Julesssss** [comment]
Looking good so far. Here are the issues I have noticed: 
- https://github.com/Expensify/App/issues/69288
    - resolved ✅ 
- https://github.com/Expensify/App/issues/69287
    - a blocker and our next priority
- https://github.com/Expensify/App/issues/69305
    - low priority UI issue in edge case flow
- [Report](https://expensify.slack.com/archives/C01GTK53T8Q/p1756236208244229) of a warning in dev: `_NativeReactNativeHybridApp.default.clearOldDotAfterSignOut is not a function `
    - Dev issue, low priority

**Comment by Julesssss** [comment]
Both deploy blockers have been fixed and Mateusz is making [progress](https://github.com/Expensify/App/issues/69305) on the remaining low priority issue that won't block deployment to prod. QA have finished testing so we're looking good!

**Comment by Julesssss** [comment]
NewDot authentication feature is on production 🤩 

Due to the size of this we had quite a few signin related bugs held on this. After retesting a good amount of these are no longer reproducible and have been closed. [Here](https://github.com/Expensify/App/issues/64348) is our current list of follow-up issues, half of which are completed with a few others in progress.

**Comment by trjExpensify** [comment]
approved $250 for @DylanDylann.
